instrumentation/elasticsearch/elasticsearch-api-client-7.16/javaagent/src/test/java/ElasticsearchClientTest.java

```diff
@@ -0,0 +1,252 @@
+
+    runWithSpan(
+        "parent",
+        () ->
+            asyncClient
+                .info()
+                .thenAccept(
+                    infoResponse ->
+                        runWithSpan(
+                            "callback",
+                            () -> {
+                              countDownLatch.countDown();
+                              request.setResponse(infoResponse);
+                            })));
+    //noinspection ResultOfMethodCallIgnored
+    countDownLatch.await(10, TimeUnit.SECONDS);
+
+    Assertions.assertEquals(request.getResponse().version().number(), "7.17.2");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("info")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "elasticsearch"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "info"),
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(SemanticAttributes.HTTP_URL, httpHost.toURI() + "/")),
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(AttributeKey.stringKey("net.protocol.name"), "http"),
+                            equalTo(AttributeKey.stringKey("net.protocol.version"), "1.1"),
+                            equalTo(SemanticAttributes.HTTP_URL, httpHost.toURI() + "/"),
+                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200L),
+                            equalTo(SemanticAttributes.USER_AGENT_ORIGINAL, userAgent()),
+                            satisfies(
+                                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                                AbstractLongAssert::isPositive)),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  private static class AsyncRequest {
+    InfoResponse response = null;
```

Comment:
Also, for thread safety, consider making this field `volatile`
```suggestion
    volatile InfoResponse response = null;
```

Reply:
done